central_v2/src/components/EGHeader.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useState } from 'react';
+import { useTheme, styled } from '@mui/material/styles';
+import { Box, Button, Typography, Collapse, IconButton } from '@mui/material';
+import rightonlogo from '../images/rightonlogo.svg';
+import dice from '../images/Dice.svg';
+import qmarks from '../images/qmarks.svg';
+import books from '../images/books.svg';
+import profile from '../images/profileplaceholder.svg';
+import hamburger from '../images/hamburger.svg';
+import hamburgerX from '../images/hamburgerX.svg';
+import plus from '../images/plus.svg';
+import { ScreenSize } from '../lib/HostModels';
+import { Screen } from '../lib/ScreenEnums';
+
+interface EGHeaderProps {
+  screenSize: ScreenSize;
+  isXLScreen: boolean;
+  onScreenChange: (newScreen: Screen) => void;
+  menuOpen: boolean;
+  setMenuOpen: (menuOpen: boolean) => void;
+}
+
+interface EGHeaderContainerProps {
+  screenSize: ScreenSize;
+  menuOpen: boolean;
+}
+const EGHeaderContainer = styled(Box)<EGHeaderContainerProps>(({ screenSize, menuOpen }) => ({
+  height: screenSize === ScreenSize.SMALL ? '77px' : '94px',
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  width: '100%',
+  padding: '16px 32px 16px 32px',
```

Comment:
So overall, the code looks really good. The styling is super clean and the composition of components is efficient. One general comment is that I'm seeing a lot of hardcoded values in the styled component definitions. Can you review these styled components and use values from our theme where possible? For example, instead of 

`padding: '16px 32px 16px 32px'`

this can be:

`padding: '${theme.sizing.smPadding}px ${theme.sizing.lgPadding}px ${theme.sizing.smPadding}px ${theme.sizing.lgPadding}px'` 

You can see `host_v2/src/components/AnswerCard.tsx` as a reference for the kinds of fields that we can substitute theme'd values for. The general rule here is if the value applies broadly, then it's good to use the theme. So the padding here is probably going to be shared across a number of components, so let's use the theme. But the height is very specific to this component itself, so we don't need a theme value for that.